### DISCUSSION TITLE
Is method chaining supported like lodash?

### DISCUSSION THREAD

**Discussion by bkkgbkjb** [rayepps/radash#224] [General]
Hi all,

I wonder if `radash` supports the same [explicit/implicit method chaining](https://stackoverflow.com/questions/35590543/how-do-you-chain-functions-using-lodash) as supported by `lodash`?

I tries in the playground but seems implicit chaining is failed:
```typescript
const _ = require('radash')

const gods = [{
  name: 'Ra',
  power: 'sun',
  rank: 100,
  culture: 'egypt'
}, {
  name: 'Loki',
  power: 'tricks',
  rank: 72,
  culture: 'norse'
}, {
  name: 'Zeus',
  power: 'lightning',
  rank: 96,
  culture: 'greek'
}]

_(gods).max(g=>g.rank)
```

with 
```shell
TypeError: _ is not a function
```

**Comment by sodiray** [comment]
This is not supported 😞 but it could be 😁 It would be pretty easy to add support for, just an object that partials all the functions and adds a `.value()` (or similar) function. I just did something similar in my exobase library. 

Two downsides I'm thinking of...

If we use `partial` we'll lose the strong typing (which is a non-starter). Maybe it's just me but I can't figure out how to type the `partial` function so that it's smart enough to return the correct argument types in the created function. Because I can't, it just returns a generic _psuedo any_ function that looses the strong typing.
```ts
const chain = <TData> (data: TData) => ({
  max: partial(max, data)
})

// Note: We would export chain as default so you could do _(gods)
chain(gods).max(g => g.rank)
```

So we'd have to _retype_ every function in the chain object. This sucks a bit because contributors/maintainers would have to make sure the chain is updated when the function actual changed.
```ts
const chain = <TData> (data: TData) => ({
  max: <T extends number | object>(
    getter?: (item: T) => number
  ) => max(data, getter)
})

chain(gods).max(g => g.rank)
```

The other downside, a bit related, is that the tests would have to be duplicated as well.

I still think chaining would be great though! I'm curious how much people want this. If a PR was made I would certainly consider it.
